include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -0,0 +1,576 @@
+                      "The flag object must be a std::UnsignedIntegral.");
+
+        static_assert(std::UnsignedIntegral<remove_cvref_t<mapq_type>>,
+                      "The mapq object must be a std::UnsignedIntegral.");
+
+        static_assert((std::ranges::ForwardRange<qual_type>       &&
+                       alphabet_concept<value_type_t<remove_cvref_t<qual_type>>>),
+                      "The qual object must be a std::ranges::ForwardRange "
+                      "over letters that model seqan3::alphabet_concept.");
+
+        static_assert(tuple_like_concept<remove_cvref_t<mate_type>>,
+                      "The mate object must be a std::tuple of size 3 with "
+                      "1) a std::ranges::ForwardRange with a value_type modelling seqan3::alphabet_concept, "
+                      "2) an std::UnsignedIntegral, and"
+                      "3) an std::UnsignedIntegral.");
+
+        static_assert((std::ranges::ForwardRange<decltype(std::get<0>(mate))>                     &&
+                      alphabet_concept<value_type_t<remove_cvref_t<decltype(std::get<0>(mate))>>> &&
+                      std::UnsignedIntegral<remove_cvref_t<decltype(std::get<1>(mate))>>          &&
+                      std::UnsignedIntegral<remove_cvref_t<decltype(std::get<2>(mate))>>),
+                      "The mate object must be a std::tuple of size 3 with "
+                      "1) a std::ranges::ForwardRange with a value_type modelling seqan3::alphabet_concept, "
+                      "2) an std::UnsignedIntegral, and"
+                      "3) an std::UnsignedIntegral.");
+
+        static_assert(std::Same<remove_cvref_t<tag_dict_type>, sam_tag_dictionary>,
+                      "The tag_dict object must be of type seqan3::sam_tag_dictionary.");
+
+        // ---------------------------------------------------------------------
+        // logical Requirements
+        // ---------------------------------------------------------------------
+        if (!ranges::empty(get<1>(align)) && ranges::empty(seq))
+            throw format_error("If you specify an align object you must also specify the seq object. "
+                               "Hint: Check if offset needs to be set to if soft-clipping is present.");
+
+        if (header_ptr != nullptr && !ranges::empty(ref_id))
+        {
+            if ((header_ptr->ref_dict).count(std::string(ref_id)) == 0) // no reference id matched
+                throw format_error(std::string("The ref_id '") + std::string(ref_id) +
+                                   "' was not in the list of references");
+        }
+
+        // ---------------------------------------------------------------------
+        // Writing the Header on first call
+        // ---------------------------------------------------------------------
+        if (!written_header && header_ptr != nullptr)
+        {
+            write_header(stream, options, header_ptr);
+            written_header = true;
+        }
```

Comment:
I think it should still throw, because it is easy to forget to edit the header. If you really don want to print it, you can set an extra option `sam_write_header = false;` as a suggested before.